client/my-sites/comments/comment-list/index.jsx

```diff
@@ -235,6 +233,25 @@ export class CommentList extends Component {
 		}
 	}
 
+	showEditNotice = ( commentId, postId, undoCommentData ) => {
```

Comment:
We've traditionally treated success and error notices as side effects to be dispatched from the `data-layer`, like:

https://github.com/Automattic/wp-calypso/blob/f17172466808e44e7b4354894c8e10ec5af1e64c/client/state/data-layer/wpcom/comments/index.js#L197

`updateComment` already has the `originalComment`

https://github.com/Automattic/wp-calypso/blob/f17172466808e44e7b4354894c8e10ec5af1e64c/client/state/data-layer/wpcom/sites/comments/index.js#L238

this makes me wonder... what would @dmsnell do? (t-shirts for sale soon)

Reply:
> what would @dmsnell do? (t-shirts for sale soon)

![http://i0.kym-cdn.com/entries/icons/mobile/000/005/574/takemymoney.jpg](http://i0.kym-cdn.com/entries/icons/mobile/000/005/574/takemymoney.jpg)

Reply:
It's true that we have the original comment in the data layer's edit comment function, exactly like we have the [previous status](https://github.com/Automattic/wp-calypso/blob/f17172466808e44e7b4354894c8e10ec5af1e64c/client/state/data-layer/wpcom/sites/comments/index.js#L33) in the change status function.
Though, it's used to revert the optimistic update on failure.

When we hit Undo, the comment in the state (and possibly in the database as well) is _already_ updated, therefore the Undo's original comment would be the _updated comment_.

Let me try an example (all made up for clarity):
- Initial state: `content: "foo"`.
- Edit the content: `editComment( "bar" )`.
- We show an optimistic success notice containing the Undo button which on click would call `editComment( "foo" )`.
- The reducer updates the comment in the Redux state as `content: "bar"`.
- The data layer creates an `originalComment = "foo"` in case of failure; then fires the API with `content: "bar"`.
- We hit Undo which calls `editComment( "foo" )`.
- The reducer updates the state to `content: "foo"`.
- The data layer creates an `originalComment = "bar"` in case of failure and fires the API with `content: "foo"`.

Now, I guess we could add something like an `isUndo` property to the `COMMENTS_EDIT` action, but I'm afraid we would do a lot of changes throughout Redux and data layer, on a flow that is not only used by us, but by other sections as well.

---

As an aside, @rodrigoi, let me say that I'm super sorry I've took control of your change status PR, ripped it apart, killed it and started over.
I wasn't familiar with the data layer at the time, and only when properly working on it I realized yours was the correct approach. 🙁 

Even so, right now we have an established (and working) pattern with the Undo controlled by the notice, which is controlled by the Comment List.
I think we should keep following that path for now, and maybe work on a refactor after M2, if we have the chance.

Reply:
@Copons don't feel sorry! I completely forgot about the problems we had in the past and the reasons behind the change. As long as we have a pattern that give the code some level of predictability, I'm OK 😉

Reply:
> I think we should keep following that path for now, and maybe work on a refactor after M2, if we have the chance.

It _does_ feel like there's opportunity for redoing the undo approach, but yeah, post-M2 seems like a better way forward.

Reply:
@dmsnell would make some 🍿 and watch the conversation

I support undoing in the response handler in order to bring it back to the pre-request state.